code/modules/mob/living/simple_animal/hostile/mining_mobs.dm

```diff
@@ -504,14 +504,40 @@ obj/item/asteroid/basilisk_hide/New()
 	if(proximity_flag && istype(target, /obj/item/clothing))
 		var/obj/item/clothing/C = target
 		var/current_armor = C.armor
+		if(!isturf(C.loc))
+			to_chat(user, "<span class='warning'>\The [C] must be safely placed on the ground for modification.</span>")
+			return
 		if(C.goliath_reinforce)
 			if(current_armor.["melee"] < 90)
 				current_armor.["melee"] = min(current_armor.["melee"] + 10, 90)
 				to_chat(user, "<span class='info'>You strengthen [target], improving its resistance against melee attacks.</span>")
 				qdel(src)
 			else
 				to_chat(user, "<span class='info'>You can't improve [C] any further.</span>")
-	return
+	if(istype(target, /obj/item/clothing/head/helmet/space/rig/mining))
```

Comment:
you don't need to do this, there's a neat proc called has_icon() that you can use to check if your suit has a goliath-icon instead